src/Main.php

```diff
@@ -24,15 +24,33 @@
 // TODO: Support running as protoc plugin.
 // TODO: Provide help/usage if incorrect command-line args provided.
 // Read command-line args.
-$opts = getopt('', ['descriptor:', 'package:']);
+// TODO: Include cmd-line arg for grpc-service-config.
+$opts = getopt('', ['descriptor:', 'package:', 'output:']);
+if (!isset($opts['descriptor']) || !isset($opts['package'])) {
+    print("Invalid arguments. Expect:\n");
+    print("  --descriptor <path> The path to the proto descriptor file.\n");
+    print("  --package <string> The proto package to generate.\n");
+    print("  --output <path> The output directory.\n");
+    print("\n");
+    exit(1);
+}
 $descBytes = stream_get_contents(fopen($opts['descriptor'], 'rb'));
 $package = $opts['package'];
+$outputDir = $opts['output'];
 
 // Generate PHP code.
 $year = (int)date('Y');
-$files = CodeGenerator::generateFromDescriptor($descBytes, $package, $year);
-foreach ($files as [$relativeFilename, $fileContent]) {
-    // TODO: Later this won't just print out the generated file content.
-    print("File: '{$relativeFilename}':\n");
-    print($fileContent . "\n");
+$files = CodeGenerator::generateFromDescriptor($descBytes, $package, $year, null);
+if (is_null($outputDir)) {
+    // TODO: Remove printout; only save files to the specified output path.
```

Comment:
But isn't this comment for the situation where we don't *have* an output path specified? It seems like a reasonable feature to keep.

Reply:
Maybe... OK, I won't remove it yet :)